Added Top videos on youtube

pankhuri813 opened the pull request:
- Added carousel for Top videos from youtube.
- Added a fetch request to get the top 10 video trending in us region from the youtube api.
- Added a toast warning for invalid playlist link.
- Added a feature to show most preffered playlist by the users.

PatelNeelMahesh commented on frontend/src/components/RadomVideos/Random.js:
```diff
@@ -0,0 +1,70 @@
+  const [videos, setVideos] = useState([]);
+
+  useEffect(() => {
+    const API_KEY = 'AIzaSyDVrPj052cWFmMOYMRoUiwJeFzArkgpwLE';
```
Add `API_KEY ` and `BASE_URL` to `.env`

PatelNeelMahesh reviewed (changes requested):
Make the required changes.

pankhuri813 commented:
- Fixed the categories pages.
- Added top 10 videos from video in cards in a different component.
- Added that page on navbar.
- fixed navbar css as well.
